### PR TITLE
shtns: init at 3.5.1

### PR DESCRIPTION
**Pull request by bzizou**
###### Motivation for this change
Library necessary for the Xshells tool (https://nschaeff.bitbucket.io/xshells/manual.html)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/physics/shtns/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ lib, stdenv, fetchgit, fftw } :
```
Comment (IvarWithoutBones) [inline pkgs/applications/science/physics/shtns/default.nix:1]:
```suggestion
{ lib, stdenv, fetchgit, fftw }:
```

File: pkgs/applications/science/physics/shtns/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "shtns";
+  version = "v3.5.1";
```
Comment (IvarWithoutBones) [inline pkgs/applications/science/physics/shtns/default.nix:5]:
```suggestion
  version = "3.5.1";
```

File: pkgs/applications/science/physics/shtns/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchgit {
+    url = "https://bitbucket.org/nschaeff/shtns";
+    rev = "${version}";
```
Comment (IvarWithoutBones) [inline pkgs/applications/science/physics/shtns/default.nix:9]:
```suggestion
    rev = "v${version}";
```
btw, you can set `rev = version;` directly when you don't need to include a versioning prefix.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]

**Comment by bobby285271** [comment]
Build fails on darwin with:

```
gcc -O2 -I/nix/store/0d9qj8xn6a20ikamnwpjy05cd1ms8aj7-shtns-3.5.1/include -L/nix/store/0d9qj8xn6a20ikamnwpjy05cd1ms8aj7-shtns-3.5.1/lib -O3 -D_GNU_SOURCE -c sht_kernels_a.c -o sht_kernels_a.o
/nix/store/rk4gf39gna5zdvvplsh5149d2c89y370-bash-5.1-p12/bin/bash: line 1: gcc: command not found
make: *** [Makefile:77: sht_kernels_a.o] Error 127
builder for '/nix/store/5bxa5g9pd3nxw1nfcc57na0h9ppnyy0b-shtns-3.5.1.drv' failed with exit code 2
```

**Comment by bzizou** [comment]
@bobby285271 I just forced-push with `platforms.linux`

**Review by IvarWithoutBones — APPROVED** [review]
LGTM :+1: 

Result of `nixpkgs-review pr 151061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>shtns</li>
  </ul>
</details>

**Pull request merged**